coralproject/talk
# Use Redis as a queue to lock stories for archiving

## What does this PR do?

Instead of searching for a story sequentially in batches which default to 500 at a time. Use Redis + Bull job system to queue up the batches prior to performing the archiving step. This allows us to query all the stories we want to archive in 1 query instead of 500 separate queries. This prevents repeat queries and eases the load on mongo to find new stories to archive which is an expensive query.

## What changes to the GraphQL/Database Schema does this PR introduce?

None

## How do I test this PR?

- set these vars in your `.env` file:

   ```
   ENABLE_AUTO_ARCHIVING=true
   AUTO_ARCHIVING_INTERVAL=0,5,10,15,20,25,30,35,40,45,50,55 * * * *
   AUTO_ARCHIVE_OLDER_THAN=60 days
   ```
- create a bunch of comments that are older than the `AUTO_ARCHIVE_OLDER_THAN`
- build and start up Coral
- watch your logs as it queues and archives the stories you created every 5 minutes on the clock
- check your db and see that stories are being archived and comments are moving over
 
## How do we deploy this PR?

Any previous or future archiving pods with `AUTO_ARCHIVING_ENABLED` set to `true` need to update their intervals as such:

```
- name: AUTO_ARCHIVING_INTERVAL
   value: "0,15,30,45 * * * *"
```
